src/Feature/Catalog/code/Controllers/ProductsSearchApiController.cs

```diff
@@ -0,0 +1,122 @@
+    using System.Collections.Specialized;
+    using System.Linq;
+    using System.Net;
+    using System.Web;
+    using System.Web.Http;
+    using System.Web.Http.Description;
+
+    using Foundation.Commerce.Models;
+    using Foundation.Commerce.Models.Catalog;
+    using Foundation.Commerce.Services.Catalog;
+    using Foundation.Commerce.Utils;
+    using Foundation.Extensions.Extensions;
+
+    [RoutePrefix(Constants.CommerceRoutePrefix + "/products/search")]
+    public class ProductsSearchApiController : ApiController
+    {
+        private readonly IProductListService productListService;
+
+        public ProductsSearchApiController(
+            IProductListService productListService)
+        {
+            this.productListService = productListService;
+        }
+
+        [ResponseType(typeof(ProductListResultModel))]
+        [HttpGet]
+        [Route("")]
+        public IHttpActionResult GetProductList(
+            [FromUri(Name = "q")] string searchKeyword = null,
+            [FromUri(Name = "pg")] int? page = null,
+            [FromUri(Name = "f")] string facetValues = null,
+            [FromUri(Name = "s")] string sortField = null,
+            [FromUri(Name = "ps")] int? pageSize = null,
+            [FromUri(Name = "sd")] SortDirection? sortDirection = null,
+            [FromUri(Name = "cci")] string currentCatalogItemId = null,
+            [FromUri(Name = "ci")] string currentItemId = null)
+        {
+            var searchParameters = this.BuildProductsSearchParametersModel(
+                searchKeyword,
+                page,
+                facetValues,
+                sortField,
+                pageSize,
+                sortDirection,
+                currentCatalogItemId,
+                currentItemId);
+
+            return this.Execute(() => this.productListService.GetProductList(searchParameters));
+        }
+
+        private ProductsSearchParametersModel BuildProductsSearchParametersModel(
+            string searchKeyword,
+            int? page,
+            string facetValues,
+            string sortField,
+            int? pageSize,
+            SortDirection? sortDirection,
+            string currentCatalogItemId,
+            string currentItemId)
+        {
+            var facetValuesCollection = !string.IsNullOrEmpty(facetValues)
+                ? HttpUtility.ParseQueryString(facetValues)
+                : new NameValueCollection();
+
+            return new ProductsSearchParametersModel()
+            {
+                CurrentCatalogItemId = currentCatalogItemId,
+                CurrentItemId = currentItemId,
+                FacetValues = facetValuesCollection,
+                PageNumber = page,
+                PageSize = pageSize,
+                SearchKeyword = searchKeyword,
+                SortDirection = sortDirection,
+                SortField = sortField,
+            };
+        }
+
+        private IHttpActionResult Execute<T>(Func<Result<T>> action)
```

Comment:
- add todo
- add task for investigating how to avoid this code duplication

for now left as is